stl/inc/expected

```diff
@@ -0,0 +1,1022 @@
+        requires is_constructible_v<_Err, initializer_list<_Uty>&, _Args...>
+    constexpr explicit unexpected(in_place_t, initializer_list<_Uty> _Ilist, _Args&&... _Vals) noexcept(
+               is_nothrow_constructible_v<_Err, initializer_list<_Uty>&, _Args...>) // strengthened
+        : _Unexpected(_Ilist, _STD forward<_Args>(_Vals)...) {}
+
+    template <class _UError = _Err>
+        requires (!is_same_v<remove_cvref_t<_UError>, unexpected> && !is_same_v<remove_cvref_t<_UError>, in_place_t> //
+                 && is_constructible_v<_Err, _UError>)
+    constexpr explicit unexpected(_UError&& _Unex) noexcept(is_nothrow_constructible_v<_Err, _UError>) // strengthened
+        : _Unexpected(_STD forward<_UError>(_Unex)) {}
+    // clang-format on
+
+    // [expected.un.observe]
+#ifdef __cpp_explicit_this_parameter // TRANSITION, P2549
+    _NODISCARD constexpr decltype(auto) error(this unexpected&& _Self) noexcept {
+        return _STD forward<unexpected>(_Self)._Unexpected;
+    }
+#else // ^^^ __cpp_explicit_this_parameter ^^^ / vvv !__cpp_explicit_this_parameter vvv
+    _NODISCARD constexpr const _Err& error() const& noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD constexpr _Err& error() & noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD constexpr const _Err&& error() const&& noexcept {
+        return _STD move(_Unexpected);
+    }
+    _NODISCARD constexpr _Err&& error() && noexcept {
+        return _STD move(_Unexpected);
+    }
+#endif // !__cpp_explicit_this_parameter
+
+    // [expected.un.swap]
+    constexpr void swap(unexpected& _Other) noexcept(is_nothrow_swappable_v<_Err>) requires is_swappable_v<_Err> {
+        _Swap_adl(_Unexpected, _Other._Unexpected);
+    }
+
+    friend constexpr void swap(unexpected& _Left, unexpected& _Right) noexcept(
+        is_nothrow_swappable_v<_Err>) requires is_swappable_v<_Err> {
+        _Left.swap(_Right);
+    }
+
+    // [expected.un.eq]
+    template <_Weakly_equality_comparable_with<_Err> _UErr>
+    _NODISCARD_FRIEND constexpr bool operator==(const unexpected& _Left, const unexpected<_UErr>& _Right) noexcept( //
+        noexcept(_Implicitly_convert_to<bool>(_Left._Unexpected == _Right.error()))) { // strengthened
+        return _Left._Unexpected == _Right.error();
+    }
+
+private:
+    _Err _Unexpected;
+};
+
+template <class _Err>
+class bad_expected_access;
+
+template <>
+class bad_expected_access<void> : public exception {
+public:
+    _NODISCARD const char* __CLR_OR_THIS_CALL what() const noexcept override {
+        return "Bad expected access";
+    }
+
+#if !_HAS_EXCEPTIONS
+protected:
+    void _Doraise() const override { // perform class-specific exception handling
+        _RAISE(*this);
+    }
+#endif // !_HAS_EXCEPTIONS
+};
+
+template <class _Err>
+class bad_expected_access : public bad_expected_access<void> {
+public:
+    __CLR_OR_THIS_CALL explicit bad_expected_access(_Err _Unex) noexcept(is_nothrow_move_constructible_v<_Err>)

```

Comment:
Only the `what()` override needs to have an explicit calling convention - this constructor doesn't. For precedent, see `bad_weak_ptr`:
https://github.com/microsoft/STL/blob/52505b9d3249c50b33eb22152a1f18f208ea2959/stl/inc/memory#L1040-L1042